RecoParticleFlow/PFClusterProducer/plugins/PFRecHitProducer.cc

```diff
@@ -54,9 +55,18 @@ void
 
    out->reserve(localRA1.upper());
    cleaned->reserve(localRA2.upper());
+
+   if (!init_) { // should go in beginRun
+     for( const auto& creator : creators_ ) {
+       creator->init(iSetup);
```

Comment:
the PFEcal{Barrel,Endcap}RecHitCreator ::init is now caching pointers to ES products. These have to be updated at IOV boundaries.
Please add a watcher
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES
 or move calls to init to every beginLumi.